alf/algorithms/rl_algorithm.py

```diff
@@ -715,20 +715,35 @@ def train_iter(self):
         else:
             return self._train_iter_off_policy()
 
-    def _train_iter_on_policy(self):
-        """User may override this for their own training procedure."""
-        alf.summary.increment_global_counter()
-
+    @data_distributed_when(lambda algorithm: algorithm.on_policy)
+    def _compute_train_info_and_loss_info_on_policy(self, unroll_length):
         with record_time("time/unroll"):
             with torch.cuda.amp.autocast(self._config.enable_amp):
                 experience = self.unroll(self._config.unroll_length)
             self.summarize_metrics()
 
+        train_info = experience.rollout_info
+        experience = experience._replace(rollout_info=())
+        loss_info = self.calc_loss(train_info)
+        self.summarize_rollout(experience)
+        return train_info, loss_info, experience
+
+    def _train_iter_on_policy(self):
+        """User may override this for their own training procedure."""
+        alf.summary.increment_global_counter()
+
+        train_info, loss_info, experience = self._compute_train_info_and_loss_info_on_policy(
+            self._config.unroll_length)
+
         with record_time("time/train"):
-            train_info = experience.rollout_info
-            experience = experience._replace(rollout_info=())
-            steps = self.train_from_unroll(experience, train_info)
```

Comment:
This seems to be the only place that `train_from_unroll` is used. Since now we are breaking it up, is there still anything preventing us from deprecating it?